src/cluster.c

```diff
@@ -4948,6 +4962,163 @@ void addNodeReplyForClusterSlot(client *c, clusterNode *node, int start_slot, in
     setDeferredArrayLen(c, nested_replylen, nested_elements);
 }
 
+void addNodeDetailsToShardReply(client *c, clusterNode *node) {
+    /*
+     * Following information regarding a node is returned.
+     *
+     * 1. id (40 char len)
+     * 2. port
+     * 3. ip
+     * 4. hostname
+     * 5. endpoint
+     * 5. role
+     * 6. replication offset (in case of replica only).
+     * 7. health
+     */
+    int reply_cnt = 0;
+    void *node_replylen = addReplyDeferredLen(c);
+    addReplyBulkCString(c, "id");
+    addReplyBulkCBuffer(c, node->name, CLUSTER_NAMELEN);
+    reply_cnt++;
+    addReplyBulkCString(c, "port");
+    addReplyLongLong(c, node->port);
+    reply_cnt++;
+    addReplyBulkCString(c, "ip");
+    addReplyBulkCString(c, node->ip);
+    reply_cnt++;
+    if (node->hostname) {
+        addReplyBulkCString(c, "hostname");
+        addReplyBulkCString(c, node->hostname);
+        reply_cnt++;
+    }
+    const char *endpoint = getPreferredEndpoint(node);
+    if (!strcmp(endpoint, "")) {
```

Comment:
I think empty string is valid for endpoint, it means "unknown endpoint".

Reply:
What's the point of unknown endpoint? Shouldn't we omit it from the map in that case?

Reply:
It means that clients should reach out to the same endpoint connected to. Omitting feels weird to me, but I suppose that could be defined behavior.

Reply:
Right, so empty string means use same as current. Omitted could mean that there is no endpoint at all, or that it's completely unknown. The word "unknown" is perhaps not the best term to use here...

Reply:
"unknown endpoint' is what we defined here, https://redis.io/commands/cluster-slots. If you have a better term we can update it, but I couldn't think of a better one.